synapse/handlers/presence.py

```diff
@@ -428,6 +433,9 @@ def user_syncing(self, user_id, affect_presence=True):
                     last_user_sync_ts=self.clock.time_msec(),
                 )])
 
+        # send any outstanding server notices to the user.
+        yield self._server_notices_sender.on_user_syncing(user_id)
```

Comment:
Do we want this here rather than in the sync handler? It feels odd to have this indirection, especially since I can see someone commenting out the presence handler in the sync code

Reply:
the reason it's here is that it takes care of the fact that we don't want to run it on the workers. Agreed that it feels odd though. Alternative solutions very welcome

Reply:
I'd be tempted to move it to sync code and then make it noop, possibly by having a dummy notice sender on workers. But not sure how worth it that is.